pkgs/by-name/mv/mvfst/package.nix

```diff
@@ -0,0 +1,149 @@
+{
+  lib,
+  stdenv,
+
+  fetchFromGitHub,
+
+  cmake,
+  ninja,
+  removeReferencesTo,
+
+  folly,
+  gflags,
+  glog,
+  apple-sdk_11,
+  darwinMinVersionHook,
+
+  fizz,
+
+  gtest,
+
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mvfst";
+  version = "2024.11.11.00";
+
+  outputs = [
+    "bin"
+    "out"
+    "dev"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "facebook";
+    repo = "mvfst";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-AQNzVe0z6psj/e4kEtjs6aHVRjqJhvkmd16XiXExmwk=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    removeReferencesTo
+  ];
+
+  buildInputs =
+    [
+      folly
+      gflags
+      glog
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      apple-sdk_11
+      (darwinMinVersionHook "11.0")
+    ];
+
+  propagatedBuildInputs = [
+    fizz
+  ];
+
+  checkInputs = [
+    gtest
+  ];
+
+  cmakeFlags =
+    [
+      (lib.cmakeBool "BUILD_SHARED_LIBS" (!stdenv.hostPlatform.isStatic))
+
+      (lib.cmakeBool "CMAKE_INSTALL_RPATH_USE_LINK_PATH" true)
+
+      (lib.cmakeBool "BUILD_TESTS" finalAttrs.finalPackage.doCheck)
+
+      (lib.cmakeFeature "CMAKE_INSTALL_PREFIX" (placeholder "dev"))
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      # Homebrew sets this, and the shared library build fails without
+      # it. I don‘t know, either. It scares me.
+      (lib.cmakeFeature "CMAKE_SHARED_LINKER_FLAGS" "-Wl,-undefined,dynamic_lookup")
+    ];
+
+  __darwinAllowLocalNetworking = true;
+
+  doCheck = true;
+
+  postPatch = ''
+    # Make sure the libraries the `tperf` binary uses are installed.
+    printf 'install(TARGETS mvfst_test_utils)\n' >> quic/common/test/CMakeLists.txt
+    printf 'install(TARGETS mvfst_dsr_backend)\n' >> quic/dsr/CMakeLists.txt
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+
+    ctest -j $NIX_BUILD_CORES --output-on-failure ${
+      lib.optionalString stdenv.hostPlatform.isLinux (
+        lib.escapeShellArgs [
+          "--exclude-regex"
+          (lib.concatMapStringsSep "|" (test: "^${lib.escapeRegex test}$") [
+            "*/QuicClientTransportIntegrationTest.NetworkTest/*"
```

Comment:
this look like globs and not a regex.

Reply:
Those are the actual test names. The regex is escaped so it matches precisely. I don’t know why it’s like that, either.

Reply:
oh yeah. I missed the escaped regex function.